include/envoy/network/listener.h

```diff
@@ -62,6 +60,36 @@ class ListenSocketFactory {
 
 using ListenSocketFactorySharedPtr = std::shared_ptr<ListenSocketFactory>;
 
+/**
+ * Configuration for a UDP listener.
+ */
+class UdpListenerConfig {
+public:
+  virtual ~UdpListenerConfig() = default;
+
+  /**
+   * @return factory for creating a listener.
+   */
+  virtual ActiveUdpListenerFactory& listenerFactory() PURE;
+
+  /**
+   * @return factory for writing to a UDP socket.
+   */
+  virtual UdpPacketWriterFactory& packetWriterFactory() PURE;
+
+  /**
+   * @return the UdpListenerWorkerRouter for this listener.
+   */
+  virtual UdpListenerWorkerRouter& listenerWorkerRouter() PURE;
+
+  /**
+   * @return the configuration for the listener.
+   */
+  virtual const envoy::config::listener::v3::UdpListenerConfig& config() PURE;
```

Comment:
IMHO all method are const methods.

Reply:
I tried going through and making all of these const changes but it's not so easy so I'm going to drop it as I don't think it actually maters.